src/java.base/windows/native/libnet/NetworkInterface.c

```diff
@@ -789,366 +326,364 @@ JNIEXPORT jobject JNICALL Java_java_net_NetworkInterface_getByName0
+    if (macAddr == NULL) {
+        return NULL;
+    }
+    (*env)->SetByteArrayRegion(
+            env, macAddr, 0, ifRow.PhysicalAddressLength,
+            (jbyte *) ifRow.PhysicalAddress);
+    return macAddr;
 }
 
 /*
  * Class:       java_net_NetworkInterface
  * Method:      getMTU0
  * Signature:   ([bLjava/lang/String;I)I
  */
-JNIEXPORT jint JNICALL Java_java_net_NetworkInterface_getMTU0
-    (JNIEnv *env, jclass class, jstring name, jint index) {
-  jint ret = -1;
-  MIB_IFROW *ifRowP;
-
-  // Retained for now to support IPv4 only stack, java.net.preferIPv4Stack
-  if (ipv6_available()) {
-    return Java_java_net_NetworkInterface_getMTU0_XP(env, class, name, index);
-  } else {
-    ifRowP = getIF(index);
-    if (ifRowP != NULL) {
-      ret = ifRowP->dwMtu;
-      free(ifRowP);
+JNIEXPORT jint JNICALL Java_java_net_NetworkInterface_getMTU0(
+        JNIEnv *env, jclass class, jstring name, jint index) {
+    MIB_IF_ROW2 ifRow = {0};
+    ULONG apiRetVal;
+
+    ifRow.InterfaceIndex = index;
+    apiRetVal = GetIfEntry2(&ifRow);
+    if (apiRetVal != NO_ERROR) {
+        SetLastError(apiRetVal);
+        NET_ThrowByNameWithLastError(
+                env, JNU_JAVANETPKG "SocketException", "GetIfEntry2");
+        return -1;
     }
-    return ret;
-  }
+    return ifRow.Mtu;
+}
+
+/*
+ * Class:     java_net_NetworkInterface
+ * Method:    supportsMulticast0
+ * Signature: (Ljava/lang/String;I)Z
+ */
+JNIEXPORT jboolean JNICALL Java_java_net_NetworkInterface_supportsMulticast0(
+        JNIEnv *env, jclass cls, jstring name, jint index) {
+    /*
+    In older versions of Windows, multicast could be disabled for individual interfaces.
+    That does not appear to be supported any longer.
+    Now, it seems to be a system-wide setting, but with separate settings for IPv4 vs IPv6.
+    We also have the additional complication that multicast sending
+    can be enabled while receiving is disabled.
+
+    There are no good ways to access these settings.
+    GetAdaptersAddresses has a NO_MULTICAST flag, but I have never seen it set,
+    even when multicast is disabled.
+    The flags appear to be stored in the registry,
+    but the exact location seems to change depending on the Windows version.
+    We could open a socket and try to join a multicast group.
+    If receiving is disabled, that fails. But if enabled, it generates network traffic.
+    I also tried some socket IOCTLs (e.g. SIO_ROUTING_INTERFACE_QUERY)
+    but all of those succeeded even when multicast was disabled.
+    Same for various setsockopt params (e.g. IP_MULTICAST_IF).
+    It's also possible to use COM to execute a WMI query:
+    "SELECT MldLevel FROM MSFT_NetIPv4Protocol" in namespace "ROOT\\StandardCimv2"
+    But during review, we agreed that wasn't great either.
+    So we're just going to unconditionally return true
+    since multicast is unlikely to be disabled,
+    and a boolean can't represent the full range of possibilities anyway.
+    */
```

Comment:
I don't think it's useful to include the historical note. Instead, I think reduce this down to a line or two to say that it optimistically assumes multicast is enabled.